File: javascript/solution.test.js
const isPalindrome = require('./solution');

describe('Palindrome', () => {
  test('simple palindromes', () => {
    expect(isPalindrome('racecar')).toBe(true);
    expect(isPalindrome('hello')).toBe(false);
  });
  test('with spaces and punctuation', () => {
    expect(isPalindrome('A man, a plan, a canal: Panama')).toBe(true);
  });
  test('empty string', () => {
    expect(isPalindrome('')).toBe(true);
  });
});

File: go/solution_test.go
package main

import "testing"

func TestIsPalindrome(t *testing.T) {
	tests := []struct {
		input    string
		expected bool
	}{
		{"racecar", true},
		{"hello", false},
		{"A man, a plan, a canal: Panama", true},
		{"", true},
		{"a", true},
		{"ab", false},
	}

	for _, tt := range tests {
		result := IsPalindrome(tt.input)
		if result != tt.expected {
			t.Errorf("IsPalindrome(%q) = %v, want %v", tt.input, result, tt.expected)
		}
	}
}
